src/equiv.js

```diff
@@ -59,6 +59,10 @@ QUnit.equiv = (function() {
 		return false;
 	}
 
+	function getRegExpFlags( regexp ) {
+		return regexp.flags || regexp.toString().match(/[gimuy]*$/)[0];
```

Comment:
The approach seems fine, but I'd prefer `"flags" in regexp ? regexp.flags : …` over `regexp.flags || …` to avoid unnecessary execution. Also, I'm not sure that `regexp.toString()` will work as we need on all environments—this needs unit tests to confirm.